src/frontend/frontend/frontend.js

```diff
@@ -650,6 +653,35 @@ angular.module('frontend-app', ['seashell-websocket', 'seashell-projects', 'jque
           self.common_files = result.common;
           self.question_files = result.question;
           self.test_files = result.tests;
+
+          self.project.currentMarmosetProject(self.question).then(function(target) {
+          if(target) {
+            marmoset.results(target).then(function(result) {
```

Comment:
Can we pull most of this (shared) code out into its own function, so we only have one copy of it in the codebase?